tier3/Dockerfile

```diff
@@ -18,6 +18,11 @@ RUN curl https://dl-ssl.google.com/linux/linux_signing_key.pub | apt-key add - &
             jq -r '[.[] | select(.prerelease | not) | .assets | flatten | .[] | select(.name | startswith("kotlin-compiler")) | .browser_download_url][0]')" && \
         unzip kotlin.zip && mv kotlinc /opt/kotlin && rm kotlin.zip && \
     curl -L -otprolog.zip "https://github.com/yingted/OpenTuring/releases/download/v1.0.0-beta/tprolog-v1.0.0-beta.zip" && \
-        unzip tprolog.zip && mv tprolog /opt && rm tprolog.zip
+        unzip tprolog.zip && mv tprolog /opt && rm tprolog.zip && \
+    mkdir /opt/zig && \
+        curl "https://ziglang.org/download/0.6.0/zig-linux-x86_64-0.6.0.tar.xz" --output zig.tar.xz && \
+        tar xf zig.tar.xz && \
+        mv zig-linux-x86_64-0.6.0/* /opt/zig && \
```

Comment:
`tar` has a flag for this, something like this should work (and also not require temporary files): https://github.com/DMOJ/runtimes-docker/blob/master/tier2/Dockerfile#L17-L19